<a href="https://www.kaggle.com/code/rstiannr/eda-cleaning-data-movie?scriptVersionId=249254511" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
# import and read file
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings("ignore")

realdf = pd.read_csv('../input/movies-dataset-for-feature-extracion-prediction/movies.csv')

# make a copy variable that use to be analysis
df = realdf.copy()

# view table
df

MOVIES         YEAR  \
0                           Blood Red Sky       (2021)   
1     Masters of the Universe: Revelation     (2021– )   
2                        The Walking Dead  (2010–2022)   
3                          Rick and Morty     (2013– )   
4                         Army of Thieves       (2021)   
...                                   ...          ...   
9994                       The Imperfects     (2021– )   
9995                               Arcane     (2021– )   
9996                    Heart of Invictus     (2022– )   
9997                       The Imperfects     (2021– )   
9998                       The Imperfects     (2021– )   

                                           GENRE  RATING  \
0         \nAction, Horror, Thriller                 6.1   
1     \nAnimation, Action, Adventure                 5.0   
2          \nDrama, Horror, Thriller                 8.2   
3     \nAnimation, Adventure, Comedy                 9.2   
4            \nAction, Crime, Horror                 NaN   
...                                          ...     ...   
9994     \nAdventure, Drama, Fantasy                 NaN   
9995  \nAnimation, Action, Adventure                 NaN   
9996            \nDocumentary, Sport                 NaN   
9997     \nAdventure, Drama, Fantasy                 NaN   
9998     \nAdventure, Drama, Fantasy                 NaN   

                                               ONE-LINE  \
0     \nA woman with a mysterious illness is forced ...   
1     \nThe war for Eternia begins again in what may...   
2     \nSheriff Deputy Rick Grimes wakes up from a c...   
3     \nAn animated series that follows the exploits...   
4     \nA prequel, set before the events of Army of ...   
...                                                 ...   
9994                                     \nAdd a Plot\n   
9995                                     \nAdd a Plot\n   
9996                                     \nAdd a Plot\n   
9997                                     \nAdd a Plot\n   
9998                                     \nAdd a Plot\n   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Gross  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN  
...    ...  
9994   NaN  
9995   NaN  
9996   NaN  
9997   NaN  
9998   NaN  

[9999 rows x 9 columns]



---

---



# Handle Duplicate


In [2]:
# check duplicate row
check_duplicate = df.duplicated().sum()
print(f"Jumlah data yang duplikat = {check_duplicate}")

Jumlah data yang duplikat = 431


Because the duplicate was in small amounts so we can drop it

In [3]:
# delete duplicate row
df= df.drop_duplicates()

df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9568 entries, 0 to 9998
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9568 non-null   object 
 1   YEAR      9026 non-null   object 
 2   GENRE     9490 non-null   object 
 3   RATING    8168 non-null   float64
 4   ONE-LINE  9568 non-null   object 
 5   STARS     9568 non-null   object 
 6   VOTES     8168 non-null   object 
 7   RunTime   7008 non-null   float64
 8   Gross     460 non-null    object 
dtypes: float64(2), object(7)
memory usage: 747.5+ KB




---

---



# Cleaning Row

The dataset contains significant noise due to the presence of extraneous characters such as parentheses "()", pipes "|", and newline symbols "\n". These irregularities disrupt the data's consistency and structure, making it difficult to analyze or process efficiently. As a result, the dataset requires thorough cleaning and pre-processing to remove these unwanted elements and ensure that the data is formatted properly for further analysis.

In [4]:
# clean year column from extract variable they doesn't need
df['YEAR'] = df['YEAR'].str.replace(r'[^0-9a-zA-Z]', '', regex=True)
df.info()

# check the latters
YearLatter = df[df['YEAR'].str.contains('[a-zA-Z]', regex=True, na=False)]
YearLatter

<class 'pandas.core.frame.DataFrame'>
Index: 9568 entries, 0 to 9998
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9568 non-null   object 
 1   YEAR      9026 non-null   object 
 2   GENRE     9490 non-null   object 
 3   RATING    8168 non-null   float64
 4   ONE-LINE  9568 non-null   object 
 5   STARS     9568 non-null   object 
 6   VOTES     8168 non-null   object 
 7   RunTime   7008 non-null   float64
 8   Gross     460 non-null    object 
dtypes: float64(2), object(7)
memory usage: 747.5+ KB


MOVIES           YEAR  \
36                Titans          I2018   
84             Heartland         II2007   
125          Escape Room          I2019   
155   Bo Burnham: Inside  2021TVSpecial   
161                   It          I2017   
...                  ...            ...   
8355          Painkiller              I   
9461          True Story            III   
9649          True Story            III   
9652          True Story            III   
9654          True Story            III   

                                        GENRE  RATING  \
36     \nAction, Adventure, Crime                 7.7   
84                \nDrama, Family                 8.4   
125   \nAction, Adventure, Horror                 6.4   
155        \nComedy, Drama, Music                 8.7   
161                      \nHorror                 7.3   
...                                       ...     ...   
8355               \nCrime, Drama                 NaN   
9461              \nComedy, Drama                 NaN   
9649              \nComedy, Drama                 NaN   
9652              \nComedy, Drama                 NaN   
9654              \nComedy, Drama                 NaN   

                                               ONE-LINE  \
36    \nA team of young superheroes combat evil and ...   
84    \nA multi-generational saga set in Alberta, Ca...   
125   \nSix strangers find themselves in a maze of d...   
155   \nA musical comedy special shot and performed ...   
161   \nIn the summer of 1989, a group of bullied ki...   
...                                                 ...   
8355                                     \nAdd a Plot\n   
9461                                     \nAdd a Plot\n   
9649                                     \nAdd a Plot\n   
9652                                     \nAdd a Plot\n   
9654                                     \nAdd a Plot\n   

                                                  STARS    VOTES  RunTime  \
36    \n            \n    Stars:\nBrenton Thwaites, ...   73,656     45.0   
84    \n            \n    Stars:\nAmber Marshall, \n...   11,764     45.0   
125   \n    Director:\nAdam Robitel\n| \n    Stars:\...   99,351     99.0   
155   \n    Director:\nBo Burnham\n| \n    Stars:\nB...   31,108     87.0   
161   \n    Director:\nAndy Muschietti\n| \n    Star...  480,607    135.0   
...                                                 ...      ...      ...   
8355                      \n    Director:\nPeter Berg\n      NaN      NaN   
9461  \n    Director:\nStephen Williams\n| \n    Sta...      NaN      NaN   
9649  \n    Director:\nStephen Williams\n| \n    Sta...      NaN      NaN   
9652  \n    Director:\nHanelle M. Culpepper\n| \n   ...      NaN      NaN   
9654  \n    Director:\nHanelle M. Culpepper\n| \n   ...      NaN      NaN   

         Gross  
36         NaN  
84         NaN  
125    $57.01M  
155        NaN  
161   $327.48M  
...        ...  
8355       NaN  
9461       NaN  
9649       NaN  
9652       NaN  
9654       NaN  

[1037 rows x 9 columns]

In [5]:
df['YEAR_len'] = df['YEAR'].str.len()
df['YEAR_len'].value_counts()

YEAR_len
4.0     6625
8.0     1366
13.0     394
5.0      172
11.0     160
9.0      119
6.0       85
7.0       30
2.0       28
3.0       24
1.0       22
10.0       1
Name: count, dtype: int64

In [6]:
# Delete the latter
df['YEAR'] = df['YEAR'].str.replace(r'[^0-9]', '', regex=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9568 entries, 0 to 9998
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9568 non-null   object 
 1   YEAR      9026 non-null   object 
 2   GENRE     9490 non-null   object 
 3   RATING    8168 non-null   float64
 4   ONE-LINE  9568 non-null   object 
 5   STARS     9568 non-null   object 
 6   VOTES     8168 non-null   object 
 7   RunTime   7008 non-null   float64
 8   Gross     460 non-null    object 
 9   YEAR_len  9026 non-null   float64
dtypes: float64(3), object(7)
memory usage: 822.2+ KB


In [7]:
# Make row with zero value be NaN
df['YEAR'] = df['YEAR'].replace('', np.nan)
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9568 entries, 0 to 9998
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9568 non-null   object 
 1   YEAR      8933 non-null   object 
 2   GENRE     9490 non-null   object 
 3   RATING    8168 non-null   float64
 4   ONE-LINE  9568 non-null   object 
 5   STARS     9568 non-null   object 
 6   VOTES     8168 non-null   object 
 7   RunTime   7008 non-null   float64
 8   Gross     460 non-null    object 
 9   YEAR_len  9026 non-null   float64
dtypes: float64(3), object(7)
memory usage: 822.2+ KB


In [8]:
# add . for year with range
def add_period(value):
    if isinstance(value, str) and len(value) == 8:
        return value[:4] + '.' + value[4:]
    return value

df['YEAR'] = df['YEAR'].apply(add_period)
df

MOVIES       YEAR  \
0                           Blood Red Sky       2021   
1     Masters of the Universe: Revelation       2021   
2                        The Walking Dead  2010.2022   
3                          Rick and Morty       2013   
4                         Army of Thieves       2021   
...                                   ...        ...   
9993                            Totenfrau       2022   
9995                               Arcane       2021   
9996                    Heart of Invictus       2022   
9997                       The Imperfects       2021   
9998                       The Imperfects       2021   

                                           GENRE  RATING  \
0         \nAction, Horror, Thriller                 6.1   
1     \nAnimation, Action, Adventure                 5.0   
2          \nDrama, Horror, Thriller                 8.2   
3     \nAnimation, Adventure, Comedy                 9.2   
4            \nAction, Crime, Horror                 NaN   
...                                          ...     ...   
9993               \nDrama, Thriller                 NaN   
9995  \nAnimation, Action, Adventure                 NaN   
9996            \nDocumentary, Sport                 NaN   
9997     \nAdventure, Drama, Fantasy                 NaN   
9998     \nAdventure, Drama, Fantasy                 NaN   

                                               ONE-LINE  \
0     \nA woman with a mysterious illness is forced ...   
1     \nThe war for Eternia begins again in what may...   
2     \nSheriff Deputy Rick Grimes wakes up from a c...   
3     \nAn animated series that follows the exploits...   
4     \nA prequel, set before the events of Army of ...   
...                                                 ...   
9993                                     \nAdd a Plot\n   
9995                                     \nAdd a Plot\n   
9996                                     \nAdd a Plot\n   
9997                                     \nAdd a Plot\n   
9998                                     \nAdd a Plot\n   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9993  \n    Director:\nNicolai Rohde\n| \n    Stars:...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Gross  YEAR_len  
0      NaN       4.0  
1      NaN       4.0  
2      NaN       8.0  
3      NaN       4.0  
4      NaN       4.0  
...    ...       ...  
9993   NaN       4.0  
9995   NaN       4.0  
9996   NaN       4.0  
9997   NaN       4.0  
9998   NaN       4.0  

[9568 rows x 10 columns]

In [9]:
# drop column we didnt need
df = df.drop('YEAR_len', axis=1)
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9568 entries, 0 to 9998
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9568 non-null   object 
 1   YEAR      8933 non-null   object 
 2   GENRE     9490 non-null   object 
 3   RATING    8168 non-null   float64
 4   ONE-LINE  9568 non-null   object 
 5   STARS     9568 non-null   object 
 6   VOTES     8168 non-null   object 
 7   RunTime   7008 non-null   float64
 8   Gross     460 non-null    object 
dtypes: float64(2), object(7)
memory usage: 747.5+ KB


In [10]:
df["ONE-LINE"] = df["ONE-LINE"].str.replace('[^a-zA-Z0-9 ]','')
df["ONE-LINE"]=df["ONE-LINE"].str.strip("\n")
df["GENRE"]=df["GENRE"].str.strip("\n")

df

MOVIES       YEAR  \
0                           Blood Red Sky       2021   
1     Masters of the Universe: Revelation       2021   
2                        The Walking Dead  2010.2022   
3                          Rick and Morty       2013   
4                         Army of Thieves       2021   
...                                   ...        ...   
9993                            Totenfrau       2022   
9995                               Arcane       2021   
9996                    Heart of Invictus       2022   
9997                       The Imperfects       2021   
9998                       The Imperfects       2021   

                                         GENRE  RATING  \
0         Action, Horror, Thriller                 6.1   
1     Animation, Action, Adventure                 5.0   
2          Drama, Horror, Thriller                 8.2   
3     Animation, Adventure, Comedy                 9.2   
4            Action, Crime, Horror                 NaN   
...                                        ...     ...   
9993               Drama, Thriller                 NaN   
9995  Animation, Action, Adventure                 NaN   
9996            Documentary, Sport                 NaN   
9997     Adventure, Drama, Fantasy                 NaN   
9998     Adventure, Drama, Fantasy                 NaN   

                                               ONE-LINE  \
0     A woman with a mysterious illness is forced in...   
1     The war for Eternia begins again in what may b...   
2     Sheriff Deputy Rick Grimes wakes up from a com...   
3     An animated series that follows the exploits o...   
4     A prequel, set before the events of Army of th...   
...                                                 ...   
9993                                         Add a Plot   
9995                                         Add a Plot   
9996                                         Add a Plot   
9997                                         Add a Plot   
9998                                         Add a Plot   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9993  \n    Director:\nNicolai Rohde\n| \n    Stars:...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Gross  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN  
...    ...  
9993   NaN  
9995   NaN  
9996   NaN  
9997   NaN  
9998   NaN  

[9568 rows x 9 columns]

The 'Stars' column in the dataset contains both directors and actors, which are currently combined in a single field. To facilitate proper analysis, this column needs to be split into two distinct columns: one for 'Directors' and another for 'Stars' (actors). This separation will enable a more structured and accurate representation of the data, allowing for easier filtering, querying, and analysis of each role independently.


In [11]:
df[['directors', 'stars']] = df['STARS'].str.split('Stars:', expand=True)
df.drop("STARS", inplace=True,axis=1)

df['directors'] = df['directors'].str.split(':', expand=True)[1]
df['directors'] = df['directors'].str.replace('|', '', regex=False)
df = df.replace(r'\n', '', regex=True)

df["VOTES"] = df["VOTES"].str.replace('[^0-9.]', '', regex=True)
df['VOTES'] = pd.to_numeric(df['VOTES'], errors='coerce')


df

MOVIES       YEAR  \
0                           Blood Red Sky       2021   
1     Masters of the Universe: Revelation       2021   
2                        The Walking Dead  2010.2022   
3                          Rick and Morty       2013   
4                         Army of Thieves       2021   
...                                   ...        ...   
9993                            Totenfrau       2022   
9995                               Arcane       2021   
9996                    Heart of Invictus       2022   
9997                       The Imperfects       2021   
9998                       The Imperfects       2021   

                                         GENRE  RATING  \
0         Action, Horror, Thriller                 6.1   
1     Animation, Action, Adventure                 5.0   
2          Drama, Horror, Thriller                 8.2   
3     Animation, Adventure, Comedy                 9.2   
4            Action, Crime, Horror                 NaN   
...                                        ...     ...   
9993               Drama, Thriller                 NaN   
9995  Animation, Action, Adventure                 NaN   
9996            Documentary, Sport                 NaN   
9997     Adventure, Drama, Fantasy                 NaN   
9998     Adventure, Drama, Fantasy                 NaN   

                                               ONE-LINE     VOTES  RunTime  \
0     A woman with a mysterious illness is forced in...   21062.0    121.0   
1     The war for Eternia begins again in what may b...   17870.0     25.0   
2     Sheriff Deputy Rick Grimes wakes up from a com...  885805.0     44.0   
3     An animated series that follows the exploits o...  414849.0     23.0   
4     A prequel, set before the events of Army of th...       NaN      NaN   
...                                                 ...       ...      ...   
9993                                         Add a Plot       NaN      NaN   
9995                                         Add a Plot       NaN      NaN   
9996                                         Add a Plot       NaN      NaN   
9997                                         Add a Plot       NaN      NaN   
9998                                         Add a Plot       NaN      NaN   

     Gross                       directors  \
0      NaN            Peter Thorwarth        
1      NaN                            None   
2      NaN                            None   
3      NaN                            None   
4      NaN      Matthias Schweighöfer        
...    ...                             ...   
9993   NaN              Nicolai Rohde        
9995   NaN                            None   
9996   NaN  Orlando von Einsiedel     Star   
9997   NaN           Jovanka Vuckovic        
9998   NaN           Jovanka Vuckovic        

                                                  stars  
0     Peri Baumeister, Carl Anton Koch, Alexander Sc...  
1     Chris Wood, Sarah Michelle Gellar, Lena Headey...  
2     Andrew Lincoln, Norman Reedus, Melissa McBride...  
3     Justin Roiland, Chris Parnell, Spencer Grammer...  
4     Matthias Schweighöfer, Nathalie Emmanuel, Ruby...  
...                                                 ...  
9993  Felix Klare, Romina Küper, Anna Maria Mühe, Ro...  
9995                                               None  
9996                                               None  
9997  Morgan Taylor Campbell, Iñaki Godoy, Rhianna J...  
9998  Morgan Taylor Campbell, Jennifer Cheon Garcia,...  

[9568 rows x 10 columns]



---

---



# Check and Analyze Missing Value

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9568 entries, 0 to 9998
Data columns (total 10 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   MOVIES     9568 non-null   object 
 1   YEAR       8933 non-null   object 
 2   GENRE      9490 non-null   object 
 3   RATING     8168 non-null   float64
 4   ONE-LINE   9568 non-null   object 
 5   VOTES      8168 non-null   float64
 6   RunTime    7008 non-null   float64
 7   Gross      460 non-null    object 
 8   directors  6304 non-null   object 
 9   stars      8365 non-null   object 
dtypes: float64(3), object(7)
memory usage: 822.2+ KB


In [13]:
df.isna().sum()

MOVIES          0
YEAR          635
GENRE          78
RATING       1400
ONE-LINE        0
VOTES        1400
RunTime      2560
Gross        9108
directors    3264
stars        1203
dtype: int64



---

---



# Handling Missing Value



Since the YEAR and GENRE columns contain categorical or fixed values that cannot be meaningfully replaced with the mean or any other imputation method, and the number of missing values in these columns is relatively small, we decided to drop the rows containing NaN values in these fields. This approach ensures that the integrity of the dataset is maintained, as replacing categorical data with arbitrary values would introduce inaccuracies, while removing only a small number of rows does not significantly affect the overall dataset. By dropping rows with missing values in these key columns, we maintain the quality of the data for subsequent analysis.

In [14]:
df = df.dropna(subset=['YEAR'])
df = df.dropna(subset=['GENRE'])

df = df.drop('Gross', axis=1)
# df = df.drop('RunTime', axis=1)

In [15]:
df.isna().sum()

MOVIES          0
YEAR            0
GENRE           0
RATING        754
ONE-LINE        0
VOTES         754
RunTime      1934
directors    2899
stars         807
dtype: int64

In [16]:
# cek statiscial summary
df.describe()

RATING         VOTES      RunTime
count  8157.000000  8.157000e+03  6977.000000
mean      6.920179  1.516472e+04    68.923033
std       1.219589  7.014462e+04    47.321144
min       1.100000  5.000000e+00     1.000000
25%       6.200000  1.670000e+02    37.000000
50%       7.100000  7.960000e+02    60.000000
75%       7.800000  3.806000e+03    95.000000
max       9.900000  1.713028e+06   853.000000

## Observations:


**RATING**

The minimum value of 1.1 and the maximum value of 9.9 make sense within the context of movie ratings, as they fall within the expected range for such scores. The median (50th percentile) value for the RATING column is 7.1, which indicates a fairly normal distribution of ratings.

**VOTES**

The minimum value of 5 and the maximum value of 1,713,028 for VOTES suggests that some movies have exceptionally high engagement, though the vast majority likely fall within a more moderate range. With a median of 796, the distribution also appears relatively balanced, though it's worth noting that the presence of extreme values (outliers) could slightly skew the mean.

**RunTime**

The minimum runtime of 1 minute and the maximum of 853 minutes seems to be reasonable. Given that 75% of the data lies below 95 minutes and the median is 60 minutes, we can expect most movies to have a runtime within a standard length. However, the high variability (standard deviation) suggests that there are some movies with significantly longer durations.

**Regarding Missing Values:**

The missing values in RATING and VOTES columns, 754 values each, can be replaced by their respective median values to ensure that the data remains consistent. The median is a robust choice for numerical columns like these, as it is less influenced by outliers compared to the mean.


In [17]:
median_runtime = df['RunTime'].median()
df['RATING'].fillna(median_runtime, inplace=True)
df['VOTES'].fillna(median_runtime, inplace=True)
df['RunTime'].fillna(median_runtime, inplace=True)

df['directors'].fillna('None', inplace=True)
df['stars'].fillna('None', inplace=True)

df

MOVIES       YEAR  \
0                           Blood Red Sky       2021   
1     Masters of the Universe: Revelation       2021   
2                        The Walking Dead  2010.2022   
3                          Rick and Morty       2013   
4                         Army of Thieves       2021   
...                                   ...        ...   
9993                            Totenfrau       2022   
9995                               Arcane       2021   
9996                    Heart of Invictus       2022   
9997                       The Imperfects       2021   
9998                       The Imperfects       2021   

                                         GENRE  RATING  \
0         Action, Horror, Thriller                 6.1   
1     Animation, Action, Adventure                 5.0   
2          Drama, Horror, Thriller                 8.2   
3     Animation, Adventure, Comedy                 9.2   
4            Action, Crime, Horror                60.0   
...                                        ...     ...   
9993               Drama, Thriller                60.0   
9995  Animation, Action, Adventure                60.0   
9996            Documentary, Sport                60.0   
9997     Adventure, Drama, Fantasy                60.0   
9998     Adventure, Drama, Fantasy                60.0   

                                               ONE-LINE     VOTES  RunTime  \
0     A woman with a mysterious illness is forced in...   21062.0    121.0   
1     The war for Eternia begins again in what may b...   17870.0     25.0   
2     Sheriff Deputy Rick Grimes wakes up from a com...  885805.0     44.0   
3     An animated series that follows the exploits o...  414849.0     23.0   
4     A prequel, set before the events of Army of th...      60.0     60.0   
...                                                 ...       ...      ...   
9993                                         Add a Plot      60.0     60.0   
9995                                         Add a Plot      60.0     60.0   
9996                                         Add a Plot      60.0     60.0   
9997                                         Add a Plot      60.0     60.0   
9998                                         Add a Plot      60.0     60.0   

                           directors  \
0               Peter Thorwarth        
1                               None   
2                               None   
3                               None   
4         Matthias Schweighöfer        
...                              ...   
9993              Nicolai Rohde        
9995                            None   
9996  Orlando von Einsiedel     Star   
9997           Jovanka Vuckovic        
9998           Jovanka Vuckovic        

                                                  stars  
0     Peri Baumeister, Carl Anton Koch, Alexander Sc...  
1     Chris Wood, Sarah Michelle Gellar, Lena Headey...  
2     Andrew Lincoln, Norman Reedus, Melissa McBride...  
3     Justin Roiland, Chris Parnell, Spencer Grammer...  
4     Matthias Schweighöfer, Nathalie Emmanuel, Ruby...  
...                                                 ...  
9993  Felix Klare, Romina Küper, Anna Maria Mühe, Ro...  
9995                                               None  
9996                                               None  
9997  Morgan Taylor Campbell, Iñaki Godoy, Rhianna J...  
9998  Morgan Taylor Campbell, Jennifer Cheon Garcia,...  

[8911 rows x 9 columns]

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8911 entries, 0 to 9998
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   MOVIES     8911 non-null   object 
 1   YEAR       8911 non-null   object 
 2   GENRE      8911 non-null   object 
 3   RATING     8911 non-null   float64
 4   ONE-LINE   8911 non-null   object 
 5   VOTES      8911 non-null   float64
 6   RunTime    8911 non-null   float64
 7   directors  8911 non-null   object 
 8   stars      8911 non-null   object 
dtypes: float64(3), object(6)
memory usage: 696.2+ KB
